packages/utils/src/utils/agent.utils.ts

```diff
@@ -38,3 +38,82 @@ export const createAgent = async ({
     shouldFetchRootKey: fetchRootKey,
   });
 };
+
+export type AgentManagerConfig = Pick<
+  CreateAgentParams,
+  "fetchRootKey" | "host"
+>;
+
+/**
+ * AgentManager class manages HttpAgent instances for different identities.
+ *
+ * It caches agents by identity to optimise resource usage and avoid unnecessary agent creation.
+ * Provides functionality to create new agents, retrieve cached agents, and clear the cache when needed.
+ */
+export class AgentManager {
+  private agents: Record<string, HttpAgent> | undefined | null = undefined;
+
+  private constructor(private readonly config: AgentManagerConfig) {
+    this.config = config;
```

Comment:
Do you need this assignation?

Isn't:

```
private constructor(private readonly config: AgentManagerConfig) {}

// or

private config: AgentManagerConfig;

private constructor(config: AgentManagerConfig) {
  this.config = config;
}
```

Reply:
completely right, it is redundant: removed!